packages/schematics/angular/application/index.ts

```diff
@@ -222,99 +378,65 @@ function addAppToWorkspaceFile(
     });
   });
 }
+
 function minimalPathFilter(path: string): boolean {
   return !path.endsWith('tsconfig.spec.json.template');
 }
 
-export default function (options: ApplicationOptions): Rule {
-  return async (host: Tree) => {
-    const appRootSelector = `${options.prefix}-root`;
-    const componentOptions: Partial<ComponentOptions> = !options.minimal
-      ? {
-          inlineStyle: options.inlineStyle,
-          inlineTemplate: options.inlineTemplate,
-          skipTests: options.skipTests,
-          style: options.style,
-          viewEncapsulation: options.viewEncapsulation,
-        }
-      : {
-          inlineStyle: options.inlineStyle ?? true,
-          inlineTemplate: options.inlineTemplate ?? true,
-          skipTests: true,
-          style: options.style,
-          viewEncapsulation: options.viewEncapsulation,
-        };
+async function getAppOptions(
+  host: Tree,
+  options: ApplicationOptions,
+): Promise<{
+  appDir: string;
+  appRootSelector: string;
+  componentOptions: Partial<ComponentOptions>;
+  folderName: string;
+  sourceDir: string;
+}> {
+  const appRootSelector = `${options.prefix}-root`;
```

Comment:
Do we have access to the value that we've used as a component selector in a template?

Reply:
I passed it in now and am using it in the index.html file